runtime/interpreter/value.go

```diff
@@ -2192,6 +2192,18 @@ func (v *ArrayValue) FirstIndex(interpreter *Interpreter, locationRange Location
 	return NilOptionalValue
 }
 
+func (v *ArrayValue) Reverse(interpreter *Interpreter, locationRange LocationRange) VoidValue {
+	count := v.Count()
+
+	for i := 0; i < count/2; i++ {
+		tempValue := v.Get(interpreter, locationRange, i)
+		v.Set(interpreter, locationRange, i, v.Get(interpreter, locationRange, count-i-1))
+		v.Set(interpreter, locationRange, count-i-1, tempValue)
```

Comment:
Wondering if we need some special handling in case something goes wrong in between the swap. 

Say the first `Set` is successful while the second `Set` panics. Not sure why that would happen but that could lead to both index `i` and `count - i - 1` having the same value.

Reply:
If something goes wrong, i.e. panics, execution aborts and all state changes are thrown away, so there is no need to revert anything manually.

Reply:
Nice, thanks for sharing.